pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -212,7 +212,8 @@ public static DistributedStagePlan constructDistributedStagePlan(DispatchableSub
       int stageId, VirtualServerAddress serverAddress) {
     return new DistributedStagePlan(stageId, serverAddress,
         dispatchableSubPlan.getQueryStageList().get(stageId).getPlanFragment().getFragmentRoot(),
-        dispatchableSubPlan.getQueryStageList().get(stageId).toStageMetadata());
+        StageMetadata.from(dispatchableSubPlan.getQueryStageList().get(stageId)),
```

Comment:
the right way to solve this is do not construct `DistributedStagePlan` on broker. instead construct the proto object directly and when deserialize, put them into multiple `DistributedStagePlan`

Reply:
e.g. intead of doing this in Dispatcher
```
            _executorService.submit(() -> client.submit(Worker.QueryRequest.newBuilder().setStagePlan(
                        QueryPlanSerDeUtils.serialize(
                            constructDistributedStagePlan(dispatchableSubPlan, finalStageId, virtualServerAddress)))
                    .putMetadata(QueryConfig.KEY_OF_BROKER_REQUEST_ID, String.valueOf(requestId))
                    .putMetadata(QueryConfig.KEY_OF_BROKER_REQUEST_TIMEOUT_MS, String.valueOf(timeoutMs))
                    .putAllMetadata(queryOptions).build(), finalStageId, queryServerInstance, deadline,
                dispatchCallbacks::offer));
```

1. directly serialized the dispatchableSubPlan to proto format:
```
QueryPlanSerDeUtils.serialize(dispatchableSubPlan, finalStageId, virtualServerAddress)
```
2. when deserialize, deserialize it into multiple distributed plan
```
List<DistributedStagePlan> QueryPlanSerDeUtils.deserialize(Worker.StagerPlan stagePlan)
```